### PR TITLE
Pubsub test

### PR DESCRIPTION
**Pull request by tgoodlet**
Adding a `trio` based pub-sub test to decipher the API and figure out the data type details.

One thing I noticed is that you have to pass `bytes` to the [message allocator in `asend()`](https://github.com/codypiersall/pynng/blob/master/pynng/_aio.py#L175).
Is this maybe something that should be encoded by default?
The error I originally got was a bit cryptic.

Also what do you think about adding this example to the docs?

### CODE REVIEW

File: pynng/_aio.py [hunk 1]
```diff
@@ -145,7 +145,10 @@ def __init__(self, obj, async_backend):
         if async_backend is None:
             async_backend = sniffio.current_async_library()
         if async_backend not in self._aio_helper_map:
-            raise ValueError('The async backend {} is not currently supported.'.format(async_backend))
+            raise ValueError(
```
Comment (tgoodlet) [inline pynng/_aio.py]: If this is too nit-picky I can just drop it.
I tend to do this kind of thing mechanically when the editor complains...

**Comment by codypiersall** [comment]
Looks good!  Just finish the comment that says `# head over to the` and I'll accept it.

It can go in the docs (maybe just a new Examples section in the README for now?), but it would need a higher level explanation about what it's doing first.

I don't want send/asend to automatically encode when they receive `str`, since pynng shouldn't really guess what encoding the data should have.

**Comment by tgoodlet** [comment]
@codypiersall as requested. Let me know if there's anything else.

If you want me to whip up docs in the PR I'm down.

**Comment by codypiersall** [comment]
Yeah.  I think the issue is that because pub/sub is unreliable, the sub sockets aren't getting the final "None" sent.  This is causing the test to hang.  You can tell it's a hanging test because the Appveyor run lasted 1 hour.

I think the best way to fix this might be to throw in a trio.sleep() in the sender before sending None for long enough to give the receivers time to consume all the messages; 100 ms would probably be plenty; 10 ms would probably be sufficient. (In a "real" application, you don't want to throw sleeps like that, but for tests it's great IMHO.)

Additionally, setting `recv_timeout` on the sub sockets would ensure that the unit test fails instead of just hanging and getting killed.  That is worth setting anyway to avoid CI timing out from happening again.

I'm hoping to be able to merge this on Wednesday or Thursday.

**Comment by codypiersall** [comment]
OK!  I'm back at my home compy.  In principle, I like this PR and planning to accept it, but the changes I mentioned above are needed:

1. Put a sleep() call to make sure that the "None" recv() is received (or some other way to ensure it happens)
2. Set recv_timeout on the sub sockets so that if something messes up in the future CI won't fail.

Thanks for the PR!  Once you've done those two things I'll accept it.

**Comment by tgoodlet** [comment]
@codypiersall cool. Yeah sorry just haven't gotten to this yet. I'll try to get this in tonight.

**Comment by codypiersall** [comment]
No worries and no rush!

**Comment by tgoodlet** [comment]
@codypiersall let's see if that works.

**Comment by codypiersall** [comment]
Looks like I was wrong about 10 ms being enough on Windows!  It probably won't be the last time I'm wrong today.

On Travis, looks like it ran into a race condition in my Message code.  You don't need to worry about that.  It

**Comment by tgoodlet** [comment]
>looks like it ran into a race condition in my Message code

Nice. Glad it caught a regression :+1:

**Comment by codypiersall** [comment]
I'll merge this next chance I get to work on this at home.  The regression was #31 almost certainly; I'll fix that soon as well.

**Pull request merged**